### ISSUE TITLE
OIDC email not being added as a claim when I decode id_token

### ISSUE THREAD

**Issue by AbulSyed** [spring-projects/spring-authorization-server#1294]
**Describe the bug**
Hi, I am trying to implement OIDC in my authorization server. I pass `openid%20email` when making the initial request to get the `authorization_code`.

After I use the `authorization_code` to obtain the `id token`. When I decode the `id token`, I don't see the email being added as a claim. Am I missing something? Do I need to manually add the email as a claim? I would think spring security would do this behind the scenes.

**Sample**

[GitHub repo](https://github.com/AbulSyed/auth-server-oidc)

**To Reproduce**
1. `docker compose up` - to run the postgres db

2. Run spring boot app using `./gradlew bootRun`

This will start the authorization server as well as create the `users` table and insert a user with credentials: username: admin, password: admin, email: a@m.com

3. Paste in browser & login with above credentials, to get the `authorization_code`

```
http://localhost:8080/oauth2/authorize?response_type=code&client_id=client&scope=openid%20email&redirect_uri=http://127.0.0.1:3000/authorized&code_challenge=QYPAZ5NU8yvtlQ9erXrUYR-T5AGCjCF47vN-KsaI2A8&code_challenge_method=S256
```

4. Make a POST request in postman (MAKE SURE TO REPLACE THE `authorization_code` BELOW after code=)

```
http://localhost:8080/oauth2/token?client_id=client&client_secret=secret&redirect_uri=http://127.0.0.1:3000/authorized&grant_type=authorization_code&code=`authorization_code`&code_verifier=qPsH306-ZDDaOE8DFzVn05TkN3ZZoVmI_6x4LsVglQI
```

5. Decode `id token` using: https://jwt.io/

**Expected behavior**
Expect the users `email` to appear as a claim when I decode the `id token`.

**Comment by jgrandja** [comment]
@AbulSyed 

> Do I need to manually add the email as a claim?

Correct. You can add the email as a claim by configuring a custom `OAuth2TokenCustomizer<JwtEncodingContext>` component. Please see the [reference documentation](https://docs.spring.io/spring-authorization-server/docs/current/reference/html/core-model-components.html#oauth2-token-customizer) on how to implement and configure.
